src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -532,6 +532,14 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 		});
 
 		// Window close
+		this._win.once('close', () => {
+			if (this._win && !this._win.isDestroyed()) {
+				this._win.webContents.session.clearStorageData({
```

Comment:
`clearStorageData` returns a promise so in order to await it properly it would have to somehow join with the lifecycle main service shutdown procedure?